File: scripts/check_twitter_status.py
"""
Command line utility to check Twitter status without the GUI
"""
import requests
import json
import sys
from datetime import datetime

# ANSI color codes for terminal output
class Colors:
    RESET = '\033[0m'
    RED = '\033[91m'
    GREEN = '\033[92m'
    YELLOW = '\033[93m'
    BLUE = '\033[94m'
    MAGENTA = '\033[95m'
    CYAN = '\033[96m'
    WHITE = '\033[97m'
    BOLD = '\033[1m'

def get_twitter_status():
    """Get Twitter status from backend"""
    try:
        print(f"{Colors.BLUE}Checking Twitter status...{Colors.RESET}")
        
        response = requests.get('http://127.0.0.1:5000/twitter/status', timeout=5)
        
        if not response.ok:
            print(f"{Colors.RED}Error: HTTP {response.status_code}{Colors.RESET}")
            print(f"{Colors.RED}Unable to fetch Twitter status. Make sure the backend server is running.{Colors.RESET}")
            return
        
        data = response.json()
        print(f"\n{Colors.BOLD}Twitter Status Report{Colors.RESET}\n")
        
        # Print API status
        status_color = Colors.YELLOW
        status_symbol = "⚠"
        
        if data.get('api_status') == "operational":
            status_color = Colors.GREEN
            status_symbol = "✓"
        elif data.get('api_status') == "error":
            status_color = Colors.RED
            status_symbol = "✗"
        
        print(f"API Status:     {status_color}{status_symbol} {data.get('api_status', 'unknown')}{Colors.RESET}")
        
        # Print rate limit info
        rate_limit = data.get('rate_limit', {})
        rate_limit_text = "Unknown"
        
        if rate_limit and rate_limit.get('remaining') != "unknown":
            rate_limit_text = f"{rate_limit.get('remaining')} calls remaining"
            
            if rate_limit.get('reset_time'):
                reset_time = datetime.fromisoformat(rate_limit['reset_time'].replace('Z', '+00:00'))
                formatted_time = reset_time.strftime('%H:%M:%S')
                rate_limit_text += f" (resets at {formatted_time})"
        
        print(f"Rate Limit:     {rate_limit_text}")
        
        # Print last thread info
        last_thread = data.get('last_thread')
        if last_thread:
            print(f"Latest Thread:  {Colors.CYAN}{last_thread}{Colors.RESET}")
        else:
            print(f"Latest Thread:  None")
        
        # Print last operation info
        operation = data.get('last_operation', {})
        if operation and operation.get('status') != "none":
            op_color = Colors.WHITE
            
            if operation.get('status') == "complete":
                op_color = Colors.GREEN
            elif operation.get('status') == "error":
                op_color = Colors.RED
            elif operation.get('status') in ["partial", "rate_limited"]:
                op_color = Colors.YELLOW
            
            print(f"Last Operation: {op_color}{operation.get('message', operation.get('status', 'unknown'))}{Colors.RESET}")
            
            if operation.get('timestamp'):
                op_time = datetime.fromisoformat(operation['timestamp'].replace('Z', '+00:00'))
                now = datetime.now()
                diff_mins = int((now - op_time).total_seconds() / 60)
                
                if diff_mins < 1:
                    time_ago = "just now"
                elif diff_mins < 60:
                    time_ago = f"{diff_mins} minute{'s' if diff_mins != 1 else ''} ago"
                else:
                    hours = diff_mins // 60
                    time_ago = f"{hours} hour{'s' if hours != 1 else ''} ago"
                
                print(f"                ({time_ago})")
        else:
            print(f"Last Operation: None")
        
    except requests.exceptions.RequestException as e:
        print(f"{Colors.RED}Error checking Twitter status: {e}{Colors.RESET}")
        print(f"{Colors.RED}Make sure the backend server is running at http://127.0.0.1:5000{Colors.RESET}")
    except Exception as e:
        print(f"{Colors.RED}Unexpected error: {e}{Colors.RESET}")

if __name__ == "__main__":
    get_twitter_status()

File: scripts/check_twitter_status.js
/**
 * Command line utility to check Twitter status without the GUI
 */
const fetch = require('node-fetch');

// ANSI color codes for terminal output
const colors = {
  reset: '\x1b[0m',
  red: '\x1b[31m',
  green: '\x1b[32m',
  yellow: '\x1b[33m',
  blue: '\x1b[34m',
  magenta: '\x1b[35m',
  cyan: '\x1b[36m',
  white: '\x1b[37m',
  bold: '\x1b[1m'
};

/**
 * Get Twitter status from backend
 */
async function getTwitterStatus() {
  try {
    console.log(`${colors.blue}Checking Twitter status...${colors.reset}`);
    
    const response = await fetch('http://127.0.0.1:5000/twitter/status');
    
    if (!response.ok) {
      console.error(`${colors.red}Error: HTTP ${response.status}${colors.reset}`);
      console.error(`${colors.red}Unable to fetch Twitter status. Make sure the backend server is running.${colors.reset}`);
      return;
    }
    
    const data = await response.json();
    console.log(`\n${colors.bold}Twitter Status Report${colors.reset}\n`);
    
    // Print API status
    let statusColor = colors.yellow;
    let statusSymbol = "⚠";
    
    if (data.api_status === "operational") {
      statusColor = colors.green;
      statusSymbol = "✓";
    } else if (data.api_status === "error") {
      statusColor = colors.red;
      statusSymbol = "✗";
    }
    
    console.log(`API Status:     ${statusColor}${statusSymbol} ${data.api_status}${colors.reset}`);
    
    // Print rate limit info
    if (data.rate_limit) {
      const rateLimit = data.rate_limit;
      let rateLimitText = "Unknown";
      
      if (rateLimit.remaining !== "unknown") {
        rateLimitText = `${rateLimit.remaining} calls remaining`;
        
        if (rateLimit.reset_time) {
          const resetTime = new Date(rateLimit.reset_time);
          const formattedTime = resetTime.toLocaleTimeString();
          rateLimitText += ` (resets at ${formattedTime})`;
        }
      }
      
      console.log(`Rate Limit:     ${rateLimitText}`);
    }
    
    // Print last thread info
    if (data.last_thread) {
      console.log(`Latest Thread:  ${colors.cyan}${data.last_thread}${colors.reset}`);
    } else {
      console.log(`Latest Thread:  None`);
    }
    
    // Print last operation info
    if (data.last_operation && data.last_operation.status !== "none") {
      const operation = data.last_operation;
      let opColor = colors.white;
      
      if (operation.status === "complete") {
        opColor = colors.green;
      } else if (operation.status === "error") {
        opColor = colors.red;
      } else if (operation.status === "partial" || operation.status === "rate_limited") {
        opColor = colors.yellow;
      }
      
      console.log(`Last Operation: ${opColor}${operation.message || operation.status}${colors.reset}`);
      
      if (operation.timestamp) {
        const opTime = new Date(operation.timestamp);
        const now = new Date();
        const diffMs = now - opTime;
        const diffMins = Math.floor(diffMs / 60000);
        
        let timeAgo;
        if (diffMins < 1) {
          timeAgo = "just now";
        } else if (diffMins < 60) {
          timeAgo = `${diffMins} minute${diffMins !== 1 ? 's' : ''} ago`;
        } else {
          const hours = Math.floor(diffMins / 60);
          timeAgo = `${hours} hour${hours !== 1 ? 's' : ''} ago`;
        }
        
        console.log(`                (${timeAgo})`);
      }
    } else {
      console.log(`Last Operation: None`);
    }
    
  } catch (error) {
    console.error(`${colors.red}Error checking Twitter status: ${error.message}${colors.reset}`);
    console.error(`${colors.red}Make sure the backend server is running at http://127.0.0.1:5000${colors.reset}`);
  }
}

// Execute the function
getTwitterStatus();
